codalab/lib/upload_manager.py

```diff
@@ -173,16 +203,38 @@ def write_git_repo(self, source: str, bundle_path: str):
             )
 
     def write_fileobj(
-        self, source_ext: str, source_fileobj: IO[bytes], bundle_path: str, unpack_archive: bool
+        self,
+        source_ext: str,
+        source_fileobj: IO[bytes],
+        bundle_path: str,
+        unpack_archive: bool,
+        bundle_conn_str=None,
+        index_conn_str=None,
+        progress_callback=None,
     ):
         if unpack_archive:
             output_fileobj = zip_util.unpack_to_archive(source_ext, source_fileobj)
         else:
             output_fileobj = GzipStream(source_fileobj)
-        # Write archive file.
+
+        # write archive file.
+        if bundle_conn_str is not None:
+            conn_str = os.environ.get('AZURE_STORAGE_CONNECTION_STRING', '')
+            os.environ['AZURE_STORAGE_CONNECTION_STRING'] = bundle_conn_str
+        bytes_uploaded = 0
```

Comment:
I would wrap this whole block in a `try:`. Then, at the very end, wrap the `os.environ['AZURE_STORAGE_CONNECTION_STRING'] = conn_str` code in a `finally:` block. This way, even if anything in this code throws an exception, we'll still eventually reset the connection string to the correct one.